### PR TITLE
tools/pyboard: Add option --no-follow

### PR DESCRIPTION
**Pull request by mbuesch**
This option helps to avoid hangs, if the board is being rebooted with --command.

Example usage:
```
python3 ./tools/pyboard.py --device /dev/ttyUSB0 --no-follow --command 'import machine; machine.reset()'
```

Without --no-follow it will wait for an infinite amount of time.

--no-follow overrides --follow.

Tested on ESP32.

### CODE REVIEW

**Comment by mikewadsten** [comment]
Should `--follow` and `--no-follow` be part of a mutually exclusive group? https://docs.python.org/3/library/argparse.html#mutual-exclusion

I'm not sure what one would expect if both are passed.

**Comment by mbuesch** [comment]
I was thinking about that, too. But I decided to allow both and give --no-follow precedence over --follow instead.
However, if you desire I can add it to an exclusive group.

File: tools/pyboard.py [hunk 1]
```diff
@@ -580,7 +585,7 @@ def execbuffer(buf):
         pyb.exit_raw_repl()
 
     # if asked explicitly, or no files given, then follow the output
-    if args.follow or (args.command is None and not args.filesystem and len(args.files) == 0):
+    if not args.no_follow and (args.follow or (args.command is None and not args.filesystem and len(args.files) == 0)):
```
Comment (dpgeorge) [inline tools/pyboard.py]: I don't think this line needs to have `no_follow` added.  This logic here will only execute if `--follow` is explicitly passed, or if nothing is passed, ie `$ pyboard.py` on its own.  So it'd only be useful for `$ pyboard.py --no-follow` which is a no-op.

File: tools/pyboard.py [hunk 2]
```diff
@@ -545,7 +546,11 @@ def main():
 
         def execbuffer(buf):
             try:
-                ret, ret_err = pyb.exec_raw(buf, timeout=None, data_consumer=stdout_write_bytes)
+                if args.no_follow:
+                    pyb.exec_raw_no_follow(buf)
+                    ret_err = None
+                else:
+                    ret, ret_err = pyb.exec_raw(buf, timeout=None, data_consumer=stdout_write_bytes)
```
Comment (dpgeorge) [inline tools/pyboard.py]: Using `no_follow` here implements more of a "detach early" operation: in normal situations the script is piped to the board, run, and then pyboard.py waits for it to finish running.  The `no_follow` option here is just forcing it to do the first part of that sequence, ie download script and run, then detach.

So it might be better to call the option `--detach-early`.... but that's possibly confusing and in the end `--no-follow` is more logical because it corresponds to `exec_raw` vs `exec_raw_no_follow`.

In summary I'm ok with this part of the PR.

**Comment by mbuesch** [comment]
Thanks for the review.
I added the options into a group and reverted the check for no-follow at the end.

**Comment by dpgeorge** [comment]
Rebased and merged in 83afd48ad93e162eec65d34c2dfe266c995fbafd
